markdown.dtx

```diff
@@ -17296,6 +17498,20 @@ larsers.PipeTable = Ct(larsers.table_row * parsers.newline
                                                     expandtabs(code))
                          end
 
+  larsers.JekyllData   = P("---")
+                       * parsers.blankline / 0
+                       * #(-parsers.blankline)  -- if followed by blank, it's an hrule
+                       * C((parsers.line - P("---") - P("..."))^0)
+                       * (P("---") + P("..."))
+                       / function(text)
+                           local tinyyaml = require("markdown-tinyyaml")
+                           data = tinyyaml.parse(text)
+                           return writer.jekyllData(data, function(s)
+                             s = s:gsub("\n", "\n\n")
```

Comment:
This is silly hack, which works around the fact that [tinyyaml](https://github.com/peposso/lua-tinyyaml) collapses several blank lines in multi-line scalar values to just a single blank line. However, this won't fly, since we want to distinguish between hard line breaks and new paragraphs. We should take this issue upstream.

Reply:
```suggestion
                             -- a hack that works around issue #13 in peposso/lua-tinyyaml
                             s = s:gsub("\n", "\n\n")
```